src/main/java/it/gov/pagopa/pu/sil/service/paymentnotification/PaymentNotificationService.java

```diff
@@ -0,0 +1,99 @@
+
+import it.gov.pagopa.paymentnotification.legacy.dto.generated.PaymentNotification;
+import it.gov.pagopa.pu.auth.dto.generated.UserInfo;
+import it.gov.pagopa.pu.debtpositions.dto.generated.DebtPositionDTO;
+import it.gov.pagopa.pu.debtpositions.dto.generated.InstallmentDTO;
+import it.gov.pagopa.pu.organization.dto.generated.OrgSilService;
+import it.gov.pagopa.pu.organization.dto.generated.Organization;
+import it.gov.pagopa.pu.sil.connector.debtpositions.DebtPositionService;
+import it.gov.pagopa.pu.sil.connector.organization.service.OrgSilServiceComponent;
+import it.gov.pagopa.pu.sil.connector.organization.service.OrganizationService;
+import it.gov.pagopa.pu.sil.connector.paymentnotification.LegacyPaymentNotificationService;
+import it.gov.pagopa.pu.sil.exception.PaymentNotFoundException;
+import it.gov.pagopa.pu.sil.mapper.PagatiMapper;
+import it.gov.pagopa.pu.sil.service.AccessTokenService;
+import it.gov.pagopa.pu.sil.service.AuthorizationService;
+import it.gov.pagopa.pu.sil.service.receipt.ReceiptService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.stereotype.Service;
+
+import java.util.Base64;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Predicate;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class PaymentNotificationService {
+  private final OrgSilServiceComponent orgSilServiceComponent;
+  private final LegacyPaymentNotificationService legacyPaymentNotificationService;
+  private final AccessTokenService accessTokenService;
+  private final OrganizationService organizationService;
+  private final DebtPositionService deptPositionService;
+  private final PagatiMapper pagatiMapper;
+  private final ReceiptService receiptService;
+
+  public void notifyPayment(Long orgSilServiceId, String nav, UserInfo loggedUser, String accessToken) {
+    OrgSilService orgSilService = orgSilServiceComponent.getOrgSilServiceById(orgSilServiceId, accessToken)
+      .orElseThrow(() -> new IllegalArgumentException("Organization service not found"));
+    AuthorizationService.validateUserForOrganizationId(orgSilService.getOrganizationId(), loggedUser);
+
+    Organization organization = organizationService.getOrganizationById(orgSilService.getOrganizationId(), accessToken)
+      .orElse(null);
+
+    List<InstallmentDTO> installmentDTOs = deptPositionService.getInstallmentsByOrganizationIdAndNav(
+      orgSilService.getOrganizationId(),
+      nav,
+      null,
+      accessToken
+    );
+
+    List<Pair<DebtPositionDTO, InstallmentDTO>> debtPositionWithInstallmentList = getDebtPositionsAndInstallments(accessToken, installmentDTOs);
+
+    PaymentNotification paymentNotification = buildPaymentNotification(debtPositionWithInstallmentList, organization, accessToken);
+
+    String silAccessToken = accessTokenService.getAccessToken(orgSilService, accessToken);
```

Comment:
can we rename this way the method?
```suggestion
    String silAccessToken = accessTokenService.getSilAccessToken(orgSilService, accessToken);
```

also the class AccessTokenService, can we rename it into SilAccessTokenService?